### PR TITLE
chore: using swc-jest instead of esbuild jest

### PR DESCRIPTION
**Pull request by chenjiahan**
# PR Details

## Description

`esbuild-jest` is no longer maintained, and have several bugs:

- https://github.com/aelbore/esbuild-jest/issues/54
- https://github.com/aelbore/esbuild-jest/issues/48

We can use `@swc/jest` to replace it.

PS: The code coverage is decreased from `60%+` to `50%+` because the coverage is calculated differently.

## Types of changes

- [x] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [ ] My code follows the code style of this project.
- [ ] I have updated changeset
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: packages/server/bff-core/tests/router.test.ts [hunk 1]
```diff
@@ -101,7 +101,7 @@ describe('test api router', () => {
     const handlerInfos = apiRouter.getSingleModuleHandlers(apiFile);
     const methods = handlerInfos?.map(handlerInfo => handlerInfo.httpMethod);
     expect(methods?.length).toBe(3);
-    expect(methods).toEqual(['DELETE', 'GET', 'PUT']);
+    expect(methods).toEqual(['GET', 'DELETE', 'PUT']);
   });
```
Comment (yimingjfe) [inline packages/server/bff-core/tests/router.test.ts:105]: 这个顺序为啥会换呀？

File: packages/server/server/tests/index.test.ts [hunk 2]
```diff
@@ -1,3 +1,6 @@
+/**
+ * @jest-environment jsdom
+ */
 import { formatURL } from '../src/hmr-client/createSocketUrl';
```
Comment (yimingjfe) [inline packages/server/server/tests/index.test.ts:4]: 这个测试文件不是测 server 的吗？这样写是为了？

Comment (chenjiahan) [reply packages/server/server/tests/index.test.ts:4]: 这个是慧琳把 hmr-client 里的代码迁移到 packages/server 下了，是一个 client 的单测，不是 server 的

Comment (chenjiahan) [reply packages/server/server/tests/index.test.ts:4]: 里面会访问 window

File: packages/server/bff-core/tests/router.test.ts (hunk 1 above)
Comment (chenjiahan) [reply packages/server/bff-core/tests/router.test.ts:105]: 我看了下 swc-jest 的顺序是对的，esbuild-jest 编译后的顺序和源代码不一致。

这个的正确性需要你再帮忙确认下~

**Review by yimingjfe — APPROVED** [review]

**Pull request merged**